### PR TITLE
file structure of the app, and the navigation.

### PR DESCRIPTION
**Pull request by sarykemmoun**

### CODE REVIEW

File: js/components/Approvals.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    render() {
+        return (
+            <Content>
+                <Text>Approvals</Text>
+            </Content>
+        );
+    }
+}
+
+export default Tickets;
```
Comment (raphaelboukara) [inline js/components/Approvals.js]:
```
function Tickets() {
    return (
        <Content>
            <Text>Approvals</Text>
        </Content>
    );
}

export default Tickets;
```
In this case you can use `stateless functional components`

File: js/components/Assistance.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    render() {
+        return (
+            <Content>
+                <Text>Assistance</Text>
+            </Content>
+        );
+    }
+}
+
+export default Tickets;
```
Comment (raphaelboukara) [inline js/components/Assistance.js]: same here

File: js/components/Login.js [hunk 3]
```diff
@@ -1,21 +1,23 @@
...
                 </Button>
             </Content>
         );
     }
 }
+Login.propTypes = {
+    onLogin: PropTypes.func
+};
 
 export default Login;
```
Comment (raphaelboukara) [inline js/components/Login.js]:
```
import React, { Component } from 'react';

import { Content, Button, Text } from 'native-base';
import PropTypes from 'prop-types';

function Login({ onLogin }) {
    return (
        <Content>
            <Button onPress={onLogin}>
                <Text>Login</Text>
            </Button>
        </Content>
    );
}

Login.propTypes = {
    onLogin: PropTypes.func.isRequired
};

export default Login;
```
same

File: js/components/News.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    render() {
+        return (
+            <Content>
+                <Text>News</Text>
+            </Content>
+        );
+    }
+}
+
+export default Tickets;
```
Comment (raphaelboukara) [inline js/components/News.js]: u mean `News` right?

Comment (raphaelboukara) [reply js/components/News.js]: same use `stateless functional components`

File: js/components/Assistance.js (hunk 2 above)
Comment (raphaelboukara) [reply js/components/Assistance.js]: you mean Assistance right?

File: js/containers/LoginContainer.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    return {
+        onLogin: () => navigation.navigate('AllTabs')
+    };
+};
+
+const LoginContainer = connect(
+    mapStateToProps
+)(Login);
+
+export default LoginContainer;
```
Comment (raphaelboukara) [inline js/containers/LoginContainer.js]:
```
const mapStateToProps = (state, { navigation }) => ({
    onLogin: () => navigation.navigate('AllTabs')
})
```

File: js/navigator/AllTabs.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    showIcon: true
+};
+
+export const AllTabs = TabNavigator({
+    Assistance,
+    Tickets,
+    Demands,
+    Approvals,
+    News
+}, { tabBarOptions });
```
Comment (raphaelboukara) [inline js/navigator/AllTabs.js]: 💯

File: js/navigator/Root.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+import Login from '../containers/LoginContainer';
+import { AllTabs } from './AllTabs';
+
+export const Root = StackNavigator({
+    Login: {
+        screen: Login,
+        navigationOptions: {
+            title: 'Login'
+        }
```
Comment (raphaelboukara) [inline js/navigator/Root.js]: you need to create the famous folder `screens` contains the `LoginScreen` contains the `LoginConnected` (in your case `LoginContainer`) component. The purpose is to decupling the business logic and the routing logic.

Comment (raphaelboukara) [reply js/navigator/Root.js]: (the `navigationOptions` will be inside the screen component)

File: js/navigator/Tabs.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import I18n from 'react-native-i18n';
+import { Icon } from 'native-base';
+
+import TicketsScreen from '../components/Tickets';
+import ApprovalsScreen from '../components/Approvals';
+import AssistanceScreen from '../components/Assistance';
+import NewsScreen from '../components/News';
```
Comment (raphaelboukara) [inline js/navigator/Tabs.js]: add al of them to screen folder

File: js/components/Approvals.js (hunk 1 above)
Comment (sarahmarciano) [reply js/components/Approvals.js]: This component is just temporary - we will need it to be a state component

File: js/components/Assistance.js (hunk 2 above)
Comment (sarahmarciano) [reply js/components/Assistance.js]: yes, indeed
a copy/paste mistake....

**Pull request closed without merging**